VideoLocker/src/main/java/org/edx/mobile/profiles/UserProfileAccomplishmentsPresenter.java

```diff
@@ -67,27 +63,15 @@ public void setProgressVisible(boolean visible) {
         }, new InfiniteScrollUtils.PageLoader<BadgeAssertion>() {
             @Override
             public void loadNextPage(@NonNull final InfiniteScrollUtils.PageLoadCallback<BadgeAssertion> callback) {
-                AsyncCallableUtils.observe(new Callable<Page<BadgeAssertion>>() {
+                userService.getBadges(username, page).enqueue(new Callback<Page<BadgeAssertion>>() {
                     @Override
-                    public Page<BadgeAssertion> call() throws Exception {
-                        return userAPI.getBadges(username, page);
-                    }
-                }, new Observer<Page<BadgeAssertion>>() {
-                    @Override
-                    public void onData(@NonNull final Page<BadgeAssertion> data) {
+                    protected void onResponse(@NonNull final Page<BadgeAssertion> badges) {
```

Comment:
Yes, the default Retrofit callback executor for the Android platform runs on the main thread.